src/mappening/api/locations.py

```diff
@@ -0,0 +1,154 @@
+          if key == "name" and 'name' in old_loc['location'] and old_loc['location']['name'] != new_loc['location']['name']:
+            if new_loc['location']['name'].lower() not in (name.lower() for name in old_loc['location']['alternative_names']):
+              if new_loc['location']['name']:
+                old_loc['location']['alternative_names'].append(new_loc['location']['name'])
+                updated = True
+            # Also add stripped down name
+            if place_name not in (name.lower() for name in old_loc['location']['alternative_names']):
+              if place_name:
+                old_loc['location']['alternative_names'].append(place_name)
+                updated = True
+
+        # Only replace document if it was updated
+        if updated:
+          updated = False
+          updated_locations.append(old_loc)
+          print "Updated: " + old_loc['location']['name']
+          # Replace document with updated info
+          if is_name:
+            UCLA_locations_collection.replace_one({'location.alternative_names': processed_place}, old_loc)  
+          else:
+            UCLA_locations_collection.replace_one({'location.latitude': new_loc['location'].get('latitude', INVALID_COORDINATE), 'location.longitude': new_loc['location'].get('longitude', INVALID_COORDINATE)}, old_loc)
+                    
+      else:
+        # No pre-existing location so insert new location to db
+        # Also add stripped version of name to location info
+        if place_name != new_loc['location']['name'].lower():
+          new_loc['location']['alternative_names'].append(place_name)
+        added_locations.append(new_loc)
+        print "Added: " + new_loc['location']['name']
+        UCLA_locations_collection.insert_one(new_loc.copy())
+
+    return jsonify({'Added Locations': added_locations, 'Updated Locations': updated_locations})
+
+# LOCATIONS SEARCH
+
+# Given location name, return location data or some indication that no location
+# could be found. Returns <num_results> top results
+@locations.route('/search/<place_query>', defaults={'num_results': None}, methods=['GET'])
+@locations.route('/search/<place_query>/<int:num_results>', methods=['GET'])
+def get_location_results(place_query, num_results):
+    search_results = location_utils.search_locations(place_query)
+
+    if not search_results:
+      return "There were no results!"
+    elif not num_results or num_results <= 0:
+      return jsonify({"Locations": search_results}) 
+    else:
+      output = []
+      for i in range(0, num_results):
+        output.append(search_results[i])
+      return jsonify({'Locations': output})
+
+# GOOGLE WRAPPER 
+# These routes are kinda long... /api/locations/google/?/<place_query>
+
+# Run Google Maps TextSearch on given query and print all results in JSON
+# Print all results in JSON, a wrapper for Google's API
+@locations.route('/google/search/text/<place_query>', methods=['GET'])
+def get_textsearch(place_query):
```

Comment:
^why routes? I imagine you dont even have to be on the server to run this

Reply:
Yeah but it is an easy way for them to get Google's places search results relevant to UCLA without having to create an app/get an api key/determine the settings. It didn't seem like it would hurt to include it but I can take it out if you want

Reply:
tis fine